core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2160,7 +2160,7 @@ class WorkspaceService(protected val userInfo: UserInfo,
     List(
       billingProjectOwnerPolicyEmail -> Set(terraBillingProjectOwnerRole, terraWorkspaceCanComputeRole),
       policyEmailsByName(SamWorkspacePolicyNames.owner) -> Set(terraWorkspaceCanComputeRole),
-      policyEmailsByName(SamWorkspacePolicyNames.canCompute) -> Set(terraWorkspaceCanComputeRole)
+      policyEmailsByName(SamWorkspacePolicyNames.canCompute) -> Set(terraWorkspaceCanComputeRole, "roles/lifesciences.workflowsRunner")
```

Comment:
I only added `roles/lifesciences.workflowsRunner` to the `canCompute` policy. I would like to understand the difference between these three policies better (`SamWorkspacePolicyNames.owner`, `SamWorkspacePolicyNames.canCompute` and `billingProjectOwnerPolicyEmail`). Would I need to add it to `billingProjectOwnerPolicyEmail` as well? cc @rtitle @dvoet

Reply:
So the keys in this map represent [Sam roles](https://github.com/broadinstitute/sam/blob/develop/src/main/resources/reference.conf#L120) on the workspace resource: billing project owner, workspace owner, can-compute user. The values in this map represent GCP IAM roles that we set in the workspace's googleProject. This is essentially saying we give project owners, workspace owners, and workspace "can-compute" users the "canCompute" custom role in the workspace's googleProject.

Based on this I think we would want to add `roles/lifesciences.workflowsRunner` to all 3 of these, assuming we want to give all 3 Sam roles the ability to launch PAPI jobs in the workspace.

Also I'm wondering if we want to put this behind a boolean config in Rawls conf. The reason to add this to Rawls (as opposed to the custom GCP role) is the flexibility to turn if off/on for specific workspaces. Maybe it would be good to start with a `addLifeSciencesWorkflowsRunnerRole` flag for now, and we could fine-tune later if necessary.

Reply:
that makes sense and based on what the 3 sam roles are, i agree that we would want to add to all three.

Also I'm wondering if we want to put this behind a boolean config in Rawls conf. The reason to add this to Rawls (as opposed to the custom GCP role) is the flexibility to turn if off/on for specific workspaces. Maybe it would be good to start with a addLifeSciencesWorkflowsRunnerRole flag for now, and we could fine-tune later if necessary.

Not entirely sure I'm following. Adding the role to the Rawls conf would mean removing it from where I'm currently adding it now so that it doesn't always get added. What I'm not sure about is how to do this conditionally based on the Rawls config

Reply:
Idk if using a conf flag will help much tho, cuz conf flag can only be flipped on/off for all in prod. This condition won't be the same as criteria for enabling it for some workspaces and not others...

Reply:
I guess I don't feel strongly on the config flag if it will just be on/off for now. Maybe just add a comment in the Rawls code for now, explaining that we're setting the lifesciences role separate from the custom role to give us flexibility in case we need to fine-tune it depending on data policies. (For example if a dataset has a regionality requirement then we would _not_ want to grant this, as it would allow users to launch compute in any region.)